tests/fast-integration/test/k8s-test.js

```diff
@@ -55,71 +76,91 @@ describe("Commerce Mock tests", function () {
     }
   });
 
-  it("should pass with ", async function () {
+  it("should pass with 😄", async function () {
     try {
-      // we can extract namespace creation into seperate step and ignore AlreadyExists error
+      console.log("Creating mocks namespace...");
+      await k8sDynamicApi.create(mocksNamespaceObj); // we can extract namespace creation into seperate step and ignore AlreadyExists error
       console.log("Creating commerce resources...");
-      await Promise.all(
-        // TODO: merge commerceObjs and appConnectorObjs into one big yaml
-        [...commerceObjs, ...appConnectorObjs].map((obj) =>
-          k8sDynamicApi.create(obj)
-        )
-      );
+      await Promise.all(commerceObjs.map((obj) => k8sDynamicApi.create(obj)));
     } catch (err) {
       expect(err.body.message).to.be.empty;
     }
 
-    await sleep(5000); // TODO: add waiting for virtualservice
-
-    console.log("Waiting for virtual service to be ready...");
     let virtualservice;
     try {
-      virtualservice = await k8sCRDApi.listNamespacedCustomObject(
-        "networking.istio.io",
-        "v1beta1",
-        "mocks",
-        "virtualservices",
-        "true",
-        undefined,
-        undefined,
-        "apirule.gateway.kyma-project.io/v1alpha1=commerce-mock.mocks"
+      console.log("Waiting for virtual service to be ready...");
+      virtualservice = await retryPromise(
```

Comment:
I would refactor the test to wrap long API invocations in some functions, example:
```
function getMockVirtualHost() {
  return k8sCRDApi
            .listNamespacedCustomObject(
              "networking.istio.io",
              "v1beta1",
              "mocks",
              "virtualservices",
              "true",
              undefined,
              undefined,
              "apirule.gateway.kyma-project.io/v1alpha1=commerce-mock.mocks"
            )
            .then((res) => {
              expect(res.body.items).to.have.lengthOf(1);
              expect(res.body.items[0]).to.have.nested.property("spec.hosts");
              expect(res.body.items[0].spec.hosts).to.have.lengthOf(1);
              expect(res.body.items[0].spec.hosts[0]).not.to.be.empty;
              return res;
            });
}

...


    let virtualservice;
    try {
      console.log("Waiting for virtual service to be ready...");
      virtualservice = await retryPromise(getMockVirtualHost, 10, 1000 );
    } catch (err) {
      expect(err.body.message).to.be.empty;
    }
```